### DISCUSSION TITLE
Support SolidStart

### DISCUSSION THREAD

**Discussion by nirtamir2** [sandulat/routes-gen#28] [Ideas]
Hi
I want to contribute to this package and support [solid-start](https://github.com/solidjs/solid-start).
I looked over your remix adapter code and found that you did not handle any regex. It is just called `remix routes` and parsing its output, which is an elegant and robust solution. Unlike remix, solid-start does not expose a command to print routes.
Given the solid-start repository, I found a way to patch it to create command like `solid-start routes` that behaves like remix routes.

I open a discussion in solid-start - maybe they can expose solid-start routes see https://github.com/solidjs/solid-start/discussions/588 

### Now if we patch the solid-start:
I notice you use yarn because we have `yarn.lock` file, but also you have `npm` in engines. What package manager + version do you use?
I use pnpm and the patch command is supported natively (although I haven't tried it on workspace's dependency)
If you use yarn v1, I think we can use `patch-package` to modify solid-start a bit and get the routes.
What you think we should do?

### Testing
What is the way you test your routes-gen? I need a way to verify the plugin adapter works.
I want to help create this plugin but I'm not sure how you want to handle the packages inside

Thanks!

**Comment by sandulat** [comment]
@nirtamir2 thank you so much for creating a PR! It would be really awesome if we could implement a `solid-start` driver!

I'd really like to avoid patching `solid-start` if it's possible to land your PR in their repo.

I'm usually using `npm link` to link the drivers locally in some of my `Remix` projects, and then I test out the CLI there.

**Comment by sandulat** [comment]
@nirtamir2 I've just seen that your PR was merged https://github.com/solidjs/solid-start/pull/609. As soon as your change gets released, let's get your PR merged into `routes-gen` as well!

**Reply by nirtamir2** [reply]
Yes, I'm waiting for the release too.
And yes, I need to change some things in this PR.
So it is not ready.
I hope this will be finished by the end of the week

**Reply by nirtamir2** [reply]
@sandulat The PR is ready (and solid-start 0.2.10 released!)

https://github.com/sandulat/routes-gen/pull/29
